### PR TITLE
#338 Edit view for roles

### PR DESCRIPTION
**Pull request by vryazanov**
#338 
* EditView for roles + tests

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#478](https://github.com/timelesslounge/timelessis/issues/478) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#478](https://github.com/timelesslounge/timelessis/issues/478) is assigned to @amihaiemil[/z](https://www.0crat.com/u/amihaiemil), here is [why](https://www.0crat.com/footprint/CF9LG1YCA/adfdfe86-1da4-4720-ba44-d413e4cb7dce); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @emilianodellacasa[/z](https://www.0crat.com/u/emilianodellacasa) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: tests/integration/it_roles_test.py [hunk 1]
```diff
@@ -17,8 +18,12 @@ def test_create(client, db_session):
...
 
 
-def test_edit(client):
-    assert client.post(url_for('role.edit', id=1)).status_code == HTTPStatus.NOT_FOUND
+def test_edit(client, db_session):
+    role = factories.RoleFactory(name="Manager")
+    client.post(url_for("role.edit", id=role.id), data={"name": "Admin"})
+
+    db_session.flush(role)
+    assert role.name == "Admin"
```
Comment (amihaiemil) [inline tests/integration/it_roles_test.py]: @vryazanov I don't understand this test, it doesn't actually edit anything, or? It just posts the role and then flushes the DB? :D

Comment (amihaiemil) [reply tests/integration/it_roles_test.py]: @vryazanov I would post (create) an original ``Role``, then post (edit) the same ``Role`` with some modifications, then simply ``query`` (using ``Role.query``) by id and see that changes are done. nothing with accessing the ``db_session`` directly :D

Comment (vryazanov) [reply tests/integration/it_roles_test.py]: @amihaiemil It creates `Role` using factories, then send new name for this role. and then just refresh `role` from DB and checks that there is correct value

Comment (vryazanov) [reply tests/integration/it_roles_test.py]: @amihaiemil in this case
`db_session.flush(role)` and `Role.query.get(role.id)` do the same thing

Comment (amihaiemil) [reply tests/integration/it_roles_test.py]: @vryazanov and is the ``fush`` necessary? I mean, does the actual ``edit`` procedure also flush the DB? In my opinion the test should never have to do additional steps (like refreshing) to see if the code under test did something (but that's just my understanding, maybe Im wrong.

Comment (vryazanov) [reply tests/integration/it_roles_test.py]: @amihaiemil anyway I'll change to `Role.query` to be more understandable. I know that `.flush` always confuses people :)

**Comment by amihaiemil** [comment]
@vryazanov PR https://github.com/timelesslounge/timelessis/pull/468 is merged, can you change the impl here to use ``role_type`` instead of name? :D

**Comment by vryazanov** [comment]
@amihaiemil updated, pls have a look

**Comment by amihaiemil** [comment]
@vryazanov nice, thanks

**Comment by amihaiemil** [comment]
@rultor good to merge

**Comment by emilianodellacasa** [comment]
@vryazanov You forgot to leave a puzzle to keep on with development of edit template using mockups

**Comment by vryazanov** [comment]
@emilianodellacasa Done, pls have a look

**Comment by emilianodellacasa** [comment]
@rultor merge

**Comment by 0crat** [comment]
Job was finished in 45 hours, bonus for fast delivery is possible (see [§36](http://www.zerocracy.com/policy.html#36))

**Comment by 0crat** [comment]
The job [#478](https://github.com/timelesslounge/timelessis/issues/478) is now out of scope

**Comment by 0crat** [comment]
Job [#478](https://github.com/timelesslounge/timelessis/issues/478) is not in the agenda of @amihaiemil[/z](https://www.0crat.com/u/amihaiemil), can't inspect

**Comment by 0crat** [comment]
@ypshenychka[/z](https://www.0crat.com/u/ypshenychka) please review this job completed by @amihaiemil[/z](https://www.0crat.com/u/amihaiemil), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @emilianodellacasa[/z](https://www.0crat.com/u/emilianodellacasa)

**Comment by ypshenychka** [comment]
@0crat quality good

**Comment by 0crat** [comment]
Order was finished, quality is "good": +25 point(s) just awarded to @amihaiemil[/z](https://www.0crat.com/u/amihaiemil)

**Comment by 0crat** [comment]
Quality review completed: +6 point(s) just awarded to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka)

**Pull request merged**